src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import typing as t
+
+from taipy.core import Core, Scenario
+from taipy.gui import Gui, State
+from taipy.gui.extension import Element, ElementLibrary, ElementProperty, PropertyType
+
+
+class GuiCoreContext:
+    _CORE_CHANGED_NAME = "core_changed"
+    _VAR_NAME = "__CCCtx"
+
+    def __init__(self, gui: Gui, core: Core) -> None:
```

Comment:
Why do you need a Core instance? It provides no service other than running a dispatcher or managing the versions. I don't see the point of embedding it in a scenario selector.